### PR TITLE
coverity fixes 2024.0.1

### PR DESCRIPTION
**Pull request by icfaust**
# Description
Fixing a default value hit for gbt classification and regression

Changes proposed in this pull request:
- add default value to gbt classification ModelImpl
- add default value to gbt regression ModelImpl

### CODE REVIEW

File: cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_model_impl.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ class ModelImpl : public daal::algorithms::gbt::classification::Model,
     typedef gbt::internal::ModelImpl ImplType;
     typedef algorithms::classifier::internal::ModelInternal ClassificationImplType;
 
-    ModelImpl(size_t nFeatures = 0) : ClassificationImplType(nFeatures) {}
+    ModelImpl(size_t nFeatures = 0) : ClassificationImplType(nFeatures), _predictionBias(0.) {}
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/dtrees/gbt/classification/gbt_classification_model_impl.h:51]:
```suggestion
    ModelImpl(size_t nFeatures = 0) : ClassificationImplType(nFeatures), _predictionBias(0.0f) {}
```

File: cpp/daal/src/algorithms/dtrees/gbt/regression/gbt_regression_model_impl.h [hunk 2]
```diff
@@ -48,7 +48,7 @@ class ModelImpl : public daal::algorithms::gbt::regression::Model,
     typedef gbt::internal::ModelImpl ImplType;
     typedef algorithms::regression::internal::ModelInternal RegressionImplType;
 
-    ModelImpl(size_t nFeatures = 0) : RegressionImplType(nFeatures) {}
+    ModelImpl(size_t nFeatures = 0) : RegressionImplType(nFeatures), _predictionBias(0.) {}
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/dtrees/gbt/regression/gbt_regression_model_impl.h:51]:
```suggestion
    ModelImpl(size_t nFeatures = 0) : RegressionImplType(nFeatures), _predictionBias(0.0f) {}
```

**Review by ahuber21 — CHANGES_REQUESTED** [review]

**Review by ahuber21 — APPROVED** [review]
Approved, assuming green CI

**Review by Vika-F — APPROVED** [review]
LGTM

**Pull request merged**